File: src/RuleEngine/Proposition.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RuleEngine
{
    public class Proposition : IRuleElement
    {
        public Proposition(string name, object value)
        {
            Name = name;
            Value = value;
        }

        public Guid Id { get; set; } = Guid.NewGuid();
        public string Name { get; set; }
        public object Value { get; set; }

        public bool Process(IRuleElement inProposition, Operator @operator)
        {
            switch (@operator.Value)
            {
                case Operators.And:
                    return And((bool)Value, (bool)inProposition.Value);
                case Operators.Not:
                    return Not((bool)Value);
                case Operators.Or:
                    return Or((bool)Value, (bool)inProposition.Value);
                case Operators.Xor:
                    return Xor((bool)Value, (bool)inProposition.Value);
                default:
                    throw new ArgumentOutOfRangeException(nameof(@operator));
            }
        }

        private bool And(bool lhs, bool rhs) => lhs && rhs;

        private bool Or(bool lhs, bool rhs) => lhs || rhs;

        private bool Xor(bool lhs, bool rhs) => lhs != rhs;

        private bool Not(bool lhs) => !lhs;
    }
}

File: src/RulesEngine.WebApp/Scripts/rules-engine/Proposition.ts
import { IRuleElement } from "./IRuleElement"
import { Operator } from "./Operator"
import { Operators } from "./Operators"

export class Proposition implements IRuleElement {
    constructor(name: string, value: any) {
        this.Name = name;
        this.Value = value;
    }

    public Id: string = "";
    public Name: string = "";
    public Value: any = {};

    Process(element: IRuleElement, operator: Operator): boolean {
        switch (operator.Value)
        {
            case Operators.And:
                return this.And(<boolean>this.Value, <boolean>element.Value);
            case Operators.Not:
                return this.Not(<boolean>this.Value);
            case Operators.Or:
                return this.Or(<boolean>this.Value, <boolean>element.Value);
            case Operators.Xor:
                return this.Xor(<boolean>this.Value, <boolean>element.Value);
            default:
                    throw new TypeError("Invalid operator supplied.");
            }
    }

    private And(lhs: boolean, rhs: boolean): boolean {
        return lhs && rhs;
    }

    private Not(lhs: boolean): boolean {
        return !lhs;
    }

    private Or(lhs: boolean, rhs: boolean): boolean {
        return lhs || rhs;
    }

    private Xor(lhs: boolean, rhs: boolean): boolean {
        return lhs != rhs;
    }
}